### ISSUE TITLE
Dynamic import from outside deno.json directory

### ISSUE THREAD

**Issue by 0xNF** [denoland/deno#27311]
> deno --version
deno 2.1.3 (stable, release, x86_64-pc-windows-msvc)
v8 13.0.245.12-rusty
typescript 5.6.2

tl;dr dynamic imports fail when importing from outside the project directory.

My simplified project structure is like so:
```
runner/
├── .vscode
│   └── launch.json
├── mylibrary
│   ├── package.json
│   └── src
│       └── index.ts
├── server
│   ├── deno.json
│   ├── deno.lock
│   └── src
│       ├── main.ts
│       └── workers
│           └── worker_main.ts
└── static
    └── components
        └── mod1
            └── mod1.ts
```

I'm trying to load user code at runtime with `import`. The contents of this code, and its location are not known at compile time, and are likely not underneath the main project directory.  

The code that runs the imports is something like this:
#### main.ts
```typescript
const location = getLocationFromConfiguration("inside"); // "C:\\Users\\me\\projects\\runner\\static\\components\\mod1\\mod1.ts";
const importlocation = `File:://${location}`; // file:///C:/Users/me/projects/runner/static/components/mod1/mod1.ts
const mod = await import(importLocation);
 ```

When launched from within vscode, aka, from the runner/ directory, this resolves fine and the code gets loaded as expected. I assume this is because static/ is part of the cwd relative directory structure.

However, if the location is instead from someplace _outside_ of the cwd structure, for example here, on the desktop, then the code fails:

#### main.ts
```typescript
const location = getLocationFromConfiguration("outside"); // "C:\\Users\\me\\desktop\\components\\mod2\\mod2.ts";
const importlocation = `File:://${location}`; // file:///C:/Users/me/desktop/components/mod2/mod2.ts
const mod = await import(importLocation); // throws ERR_MODULE_NOT_FOUND
 ```

Using the `import.meta.resolve` method, the resolved address for the module import seems to be this:
```typescript
const p = import.meta.resolve(importlocation); 
console.log(p); // 'file:///C:/Users/me/projects/runner/Users/me/desktop/components/mod2/mod2.ts'
```
which is clearly incorrect, and is obvious why the import fails to find the module.

I don't know anything about how importing is supposed to work, but given that I gave an absolute path on disk, I expected Deno to read the import from that absolute path, instead of deciding it was a relative import.

**Comment by dsherret** [comment]
Do you have some reproduction code? This seems incorrect for example: `File:://${location}`

Can you try using a better way for converting paths to urls? For example, using https://jsr.io/@std/path@1.0.8/doc/~/toFileUrl or https://nodejs.org/api/url.html#url_url_pathtofileurl_path?

**Comment by 0xNF** [comment]
Ah, you're right. I didn't realize paths were so fraught. Many of the tutorials I've seen probably didn't consider working with Windows paths. For my part, this is the first time I'm seeing those PathToFileURL apis, thanks for pointing those out.

Things are working well now with the following code: 

```typescript
/** Takes a path, for example, "C:\\Users\\me\\desktop\\components\\mod2\\mod2.ts" and imports it. */
export async function dynamicImport(location: string): Promise<any> {
  const filePath = fileURLToPath(`File:///${location}`);
  const p = import.meta.resolve(filePath);
  const mod = await import(`File:///${p}`);
  return mod;
}

const location = "C:\\Users\\me\\desktop\\components\\mod2\\mod2.ts"
const m = await dynamicImport(location); // a fully resolved imported module.
```

I don't know why I have to keep adding the `File` portion manually since it keeps getting stripped out, but it's working fine otherwise. Thanks.

**Comment by dsherret** [comment]
I think you probably want to do:

```ts
import { pathToFileURL } from "node:url";

export async function dynamicImport(filePath: string): Promise<any> {
  const url = pathToFileURL(filePath);
  return await import(p);
}

const location = "C:\\Users\\me\\desktop\\components\\mod2\\mod2.ts"
const m = await dynamicImport(location); // a fully resolved imported module.
```

> Many of the tutorials I've seen probably didn't consider working with Windows paths.

Yeah, a lot of code out there doesn't take into account Windows properly.
